### PR TITLE
SUBS-641

### PR DESCRIPTION
**Pull request by fpenim**
`ValidationResult` and `SingleValidationResult` restructure with some breaking changes.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/validator/data/SingleValidationResult.java [hunk 1]
```diff
@@ -1,46 +1,28 @@
...
-    public SingleValidationResult() {
-    }
+    private SingleValidationResultStatus validationStatus = SingleValidationResultStatus.Pending;
 
-    public SingleValidationResult(ValidationAuthor validationAuthor, String entityUuid) {
-        this.validationAuthor = validationAuthor;
-        this.setEntityUuid(entityUuid);
-        this.setValidationStatus(ValidationStatus.Pending);
-    }
+    private String message;
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/data/SingleValidationResult.java]: I can't see any usage of this private field

File: src/main/java/uk/ac/ebi/subs/validator/data/SingleValidationResult.java [hunk 2]
```diff
@@ -1,46 +1,28 @@
...
 
-    @Override
-    public int getVersion() {
-        return version;
-    }
+    private String entityUuid;
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/data/SingleValidationResult.java]: Do we need accessors for this field?

File: src/main/java/uk/ac/ebi/subs/validator/data/structures/SingleValidationResultStatus.java [hunk 3]
```diff
@@ -1,25 +1,19 @@
...
+ * Controlled vocabulary for single validation result status
  *
  * <li>{@link #Pending}</li>
  * <li>{@link #Pass}</li>
  * <li>{@link #Error}</li>
  * <li>{@link #Warning}</li>
- * <li>{@link #Complete}</li>
  */
-public enum ValidationStatus {
+public enum SingleValidationResultStatus {
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/data/structures/SingleValidationResultStatus.java:11]: You could remove the unneeded statuses from the JavaDoc

**Review by ke4 — CHANGES_REQUESTED** [review]
Small things, but generally fine 👍

File: src/main/java/uk/ac/ebi/subs/validator/data/structures/SingleValidationResultStatus.java (hunk 3 above)
Comment (fpenim) [reply src/main/java/uk/ac/ebi/subs/validator/data/structures/SingleValidationResultStatus.java:11]: There's no unneeded statuses here: Pending, Pass, Error and Warning.

File: src/main/java/uk/ac/ebi/subs/validator/data/SingleValidationResult.java (hunk 1 above)
Comment (fpenim) [reply src/main/java/uk/ac/ebi/subs/validator/data/SingleValidationResult.java]: It's used by the validators to set the error message

File: src/main/java/uk/ac/ebi/subs/validator/data/structures/SingleValidationResultStatus.java (hunk 3 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/validator/data/structures/SingleValidationResultStatus.java:11]: Sorry, I missed this one.

**Review by ke4 — APPROVED** [review]

**Pull request merged**